### PR TITLE
Fix Positioning of Control Header on Google Desktop

### PR DESCRIPTION
**Pull request by aug-dev**
# Summary

This PR fixes an issue reported on #463, where the uBlacklist control header bar (the one that shows how many sites were blocked) would overlap with the Google "Tools" bar.

## Demo

[control-header-demo.webm](https://github.com/iorate/ublacklist/assets/109992671/44df5a4c-8674-42d5-b5d8-8a6fe54d8f7a)

[Screencast from 2024-05-26 22-24-50.webm](https://github.com/iorate/ublacklist/assets/109992671/f1cb1665-c7a1-4e5f-af8f-3843dcfc30a4)

## Small Note

There is a very small use of the `:has()` selector here:

https://github.com/iorate/ublacklist/blob/af17cb81ab005b7199e0f7bb94a7d1626539b3b1/src/scripts/search-engines/google-desktop.ts#L158-L161

This is a rule that removes a little bit of the margin that was added when there is no blocked entries.

I've tested it on Firefox ESR, and everything is fine.

This is pretty much a case of [graceful degradation](https://en.wikipedia.org/wiki/Fault_tolerance).

### CODE REVIEW

**Comment by aug-dev** [comment]
This PR was extended so now it also provides a solution to issue #475.

**Review by iorate — COMMENTED** [review]
Thank you always for your contribution.
Please check my comment.

File: src/scripts/search-engines/google-desktop.ts [hunk 1]
```diff
@@ -100,6 +101,48 @@ const regularEntryHandler: Pick<
...
 
+const regularControlHandlers: ControlHandler[] = [
+  {
+    target: "#result-stats",
+    style: {
+      // Displays on the next line when part of the "Tools" bar
+      "#hdtbMenus :is(.BfdGL.ZkEmPc + div) &": {
+        display: "block",
+      },
+      ":not(#hdtbMenus) &": {
```
[the remark this answers is not in the document]
Comment (aug-dev) [reply src/scripts/search-engines/google-desktop.ts]: Good catch! 

Here's the fixed rule:

https://github.com/iorate/ublacklist/blob/7714cde8dfa99314c49f528190bf9aeff91cb7c8/src/scripts/search-engines/google-desktop.ts#L112-L114

**Comment by aug-dev** [comment]
Unfortunately I've found some issues in certain result pages:

image

I will fix it, but until then, let's put this PR on hold.

**Comment by aug-dev** [comment]
image

Fixed.

**Review by iorate — APPROVED** [review]

**Pull request merged**